File: examples/dotnet/SendPush.cs
using System;
using System.Threading.Tasks;
using Pushfy;

// Create and send a Push Notifications campaign via the server API.
// The Push server endpoints are HMAC-signed, so this needs PushKey/PushSecret.
//   export PUSHFY_PUSH_KEY=pushk_...
//   export PUSHFY_PUSH_SECRET=pss_...
//   dotnet run
internal static class SendPushExample
{
    public static async Task Main()
    {
        var pushKey = Environment.GetEnvironmentVariable("PUSHFY_PUSH_KEY");
        var pushSecret = Environment.GetEnvironmentVariable("PUSHFY_PUSH_SECRET");
        if (string.IsNullOrEmpty(pushKey) || string.IsNullOrEmpty(pushSecret))
        {
            Console.Error.WriteLine("Set PUSHFY_PUSH_KEY and PUSHFY_PUSH_SECRET in your environment.");
            Environment.Exit(2);
        }

        using var pushfy = new PushfyClient(new PushfyClientOptions
        {
            PushKey = pushKey,
            PushSecret = pushSecret,
        });

        try
        {
            var campaign = await pushfy.Push.Campaigns.CreateAsync(new
            {
                name = "Promo July",
                title = "Sale!",
                body = "50% off today only",
                url = "https://example.com/sale",
            });
            var id = campaign.GetProperty("id").GetInt32();
            Console.WriteLine($"Created campaign {id}");

            await pushfy.Push.Campaigns.SendAsync(id);
            Console.WriteLine($"Campaign {id} sent.");

            var metrics = await pushfy.Push.Campaigns.MetricsAsync(id);
            Console.WriteLine("Metrics: " + metrics);
        }
        catch (RateLimitException err)
        {
            Console.Error.WriteLine($"Rate limited. Retry after {err.RetryAfter}s.");
            Environment.Exit(1);
        }
        catch (AuthenticationException err)
        {
            Console.Error.WriteLine($"Auth failed ({err.Status}). Check push key/secret.");
            Environment.Exit(1);
        }
        catch (ApiException err)
        {
            Console.Error.WriteLine($"Transient failure: {err.Status} {err.Message}");
            Environment.Exit(1);
        }
    }
}

File: examples/java/SendPush.java
import com.pushfy.ApiException;
import com.pushfy.AuthenticationException;
import com.pushfy.Pushfy;
import com.pushfy.RateLimitException;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Create and send a Push Notifications campaign through the server API.
 *
 * <p>The Push server endpoints are HMAC-signed, so this needs {@code pushKey} /
 * {@code pushSecret} rather than the messaging Bearer token. Set:
 * <pre>
 *   export PUSHFY_PUSH_KEY=pushk_...
 *   export PUSHFY_PUSH_SECRET=pss_...
 * </pre>
 */
public class SendPush {

    public static void main(String[] args) {
        String pushKey = System.getenv("PUSHFY_PUSH_KEY");
        String pushSecret = System.getenv("PUSHFY_PUSH_SECRET");
        if (pushKey == null || pushSecret == null) {
            System.err.println("Set PUSHFY_PUSH_KEY and PUSHFY_PUSH_SECRET in your environment.");
            System.exit(2);
        }

        Pushfy pushfy = Pushfy.builder()
                .pushKey(pushKey)
                .pushSecret(pushSecret)
                .build();

        Map<String, Object> body = new LinkedHashMap<>();
        body.put("name", "Promo July");
        body.put("title", "Sale!");
        body.put("body", "50% off today only");
        body.put("url", "https://example.com/sale");

        try {
            Object campaign = pushfy.push.campaigns.create(body);
            Object id = ((Map<?, ?>) campaign).get("id");
            System.out.println("Created campaign " + id);

            pushfy.push.campaigns.send(id);
            System.out.println("Campaign " + id + " sent.");

            Object metrics = pushfy.push.campaigns.metrics(id);
            System.out.println("Metrics: " + metrics);
        } catch (RateLimitException err) {
            System.err.println("Rate limited. Retry after " + err.retryAfter + "s.");
            System.exit(1);
        } catch (AuthenticationException err) {
            System.err.println("Auth failed (" + err.status + "). Check push key/secret.");
            System.exit(1);
        } catch (ApiException err) {
            System.err.println("Transient failure: " + err.status + " " + err.getMessage());
            System.exit(1);
        }
    }
}
